internal/resource/url.go

```diff
@@ -319,6 +332,41 @@ func (f *Fetcher) fetchFromDataURL(u url.URL, dest io.Writer, opts FetchOptions)
 	return f.decompressCopyHashAndVerify(dest, bytes.NewBuffer(url.Data), opts)
 }
 
+// FetchFromGCS writes the data stored in a GCS bucket as described by u into dest,returning
+// an error if one is encountered. It looks for the default credentials if available, else
+// will fetch the object content with anonymous credentials.
+func (f *Fetcher) fetchFromGCS(u url.URL, dest io.Writer, opts FetchOptions) error {
+	ctx := context.Background()
+	creds, err := google.FindDefaultCredentials(ctx, storage.ScopeReadOnly)
+	if err != nil {
+		if err.Error() == ErrDefaultAuthenticationForGCP.Error() {
```

Comment:
> It might be reasonable to just reimplement the relevant part of FindDefaultCredentials(), since it seems to use entirely public API

I like this idea but I have one follow-up question on that:  Do we need to consider a case where a user will initialize an environment variable `GOOGLE_APPLICATION_CREDENTIALS` to fetch the default credentials?